src/main/java/io/cdap/plugin/http/source/batch/HttpBatchSourceConfig.java

```diff
@@ -24,4 +24,23 @@
   protected HttpBatchSourceConfig(String referenceName) {
     super(referenceName);
   }
+
+  public HttpBatchSourceConfig(String referenceName, String url, String httpMethod, String headers,
```

Comment:
This can be package private. Also annotate with @VisibleForTesting

Reply:
added annotation.
changing access specifier, sorry I didn't get this because its being called from test package

Reply:
I would rather prefer to have a public builder and private constructor, which is the practice we follow.

Reply:
👍 yes we can have a builder

Reply:
@rmstar Please confirm

Reply:
Please follow @itsankit-google 's suggestion